server/datastore/mysql/software.go

```diff
@@ -2877,301 +2910,536 @@ last_vpp_install AS (
+			bySoftwareTitleID = tempBySoftwareTitleID
+			byVPPAdamID = tmpByVPPAdamID
+		}
+	}
 
-	// perform a second query to grab the titleCount
-	var titleCount uint
-	if err := sqlx.GetContext(ctx, ds.reader(ctx), &titleCount, countStmt, args...); err != nil {
-		return nil, nil, ctxerr.Wrap(ctx, err, "get host software count")
-	}
-
-	type hostSoftware struct {
-		fleet.HostSoftwareWithInstaller
-		LastInstallInstalledAt         *time.Time `db:"last_install_installed_at"`
-		LastInstallInstallUUID         *string    `db:"last_install_install_uuid"`
-		LastUninstallUninstalledAt     *time.Time `db:"last_uninstall_uninstalled_at"`
-		LastUninstallScriptExecutionID *string    `db:"last_uninstall_script_execution_id"`
-		PackageSelfService             *bool      `db:"package_self_service"`
-		PackageName                    *string    `db:"package_name"`
-		PackageVersion                 *string    `db:"package_version"`
-		PackagePlatform                *string    `db:"package_platform"`
-		VPPAppSelfService              *bool      `db:"vpp_app_self_service"`
-		VPPAppAdamID                   *string    `db:"vpp_app_adam_id"`
-		VPPAppVersion                  *string    `db:"vpp_app_version"`
-		VPPAppPlatform                 *string    `db:"vpp_app_platform"`
-		VPPAppIconURL                  *string    `db:"vpp_app_icon_url"`
+	var softwareTitleIds []uint
+	for softwareTitleID := range bySoftwareTitleID {
+		softwareTitleIds = append(softwareTitleIds, softwareTitleID)
 	}
-	var hostSoftwareList []*hostSoftware
-	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &hostSoftwareList, stmt, args...); err != nil {
-		return nil, nil, ctxerr.Wrap(ctx, err, "list host software")
+
+	var softwareIDs []uint
+	for softwareID := range bySoftwareID {
+		softwareIDs = append(softwareIDs, softwareID)
 	}
 
-	// collect the title ids to get the versions, vulnerabilities and installed
-	// paths for each software in the list.
-	titleIDs := make([]uint, 0, len(hostSoftwareList))
-	byTitleID := make(map[uint]*hostSoftware, len(hostSoftwareList))
-	for _, hs := range hostSoftwareList {
+	var vppAdamIDs []string
+	for key := range byVPPAdamID {
+		vppAdamIDs = append(vppAdamIDs, key)
+	}
 
-		// promote the package name and version to the proper destination fields
-		if hs.PackageName != nil {
-			var version string
-			if hs.PackageVersion != nil {
-				version = *hs.PackageVersion
-			}
-			var platform string
-			if hs.PackagePlatform != nil {
-				platform = *hs.PackagePlatform
+	var titleCount uint
+	var hostSoftwareList []*hostSoftware
+	if len(softwareTitleIds) > 0 || len(vppAdamIDs) > 0 {
+		var args []interface{}
+		var stmt string
+		var softwareTitleStatment string
+		var vppAdamStatment string
+
+		matchClause := ""
+		matchArgs := []interface{}{}
+		if opts.ListOptions.MatchQuery != "" {
+			matchClause, matchArgs = searchLike(matchClause, matchArgs, opts.ListOptions.MatchQuery, "software_titles.name", "software_cve.cve")
```

Comment:
This was really the only modification to add matching by CVE name for the ticket
`+ , "software_cve.cve"`